store/signature_request.go

```diff
@@ -15,12 +15,14 @@ const (
 type SignatureRequest struct {
 	IncomingType            NetworkType `db:"incoming_type"`
 	IncomingTransactionHash string      `db:"incoming_transaction_hash"`
+	TxExpirationTimestamp   int64       `db:"tx_expiration_timestamp"`
```

Comment:
Is this set by the bridge contract on Ethereum? Another option would be to expose the timestamp of the block in which the deposit transaction appears (let's call it T). Bridge validators can then be set up to make the outgoing Stellar transaction expire at T+Delta, where Delta is configurable.

Reply:
No, this will be a timestamp provided by the client (either local time or last ledger close time seen by the client - both plus a few minutes). The reason is that different Starbridge validators can be at different ledgers when processing the request so the final transaction will be different on some validators so a signature will be invalid.

Reply:
Okay, makes sense. If we don't want to leave it to the client, block time (of the incoming transaction, which is uniquely determined) + delta would also work, as all validators would compute the same max time.

Reply:
I think there may be a misunderstanding here (not sure if on my side or yours). `TxExpirationTimestamp` is a `timebounds.max_time` proposed by a client (and validated by Starbridge before signing). We can't use incoming tx block time because there can be multiple withdrawal expired/invalid transactions.

Reply:
Okay, that makes sense. I thought we could keep increasing by delta (e.g. the first time it would be T+Delta, and if the client requests a second outgoing transaction then T+2Delta, etc.) but there's a bunch of issues with that (e.g. with the deltas there would be problems if some validators are not aware of all past outgoing tx requests) and so I agree it's better for the client to set this instead (and for starbridge to just validate it).